compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerationExtension.kt

```diff
@@ -70,12 +72,16 @@ internal class CodeGenerationExtension(
 
     val anvilContext = commandLineOptions.toAnvilContext(anvilModule)
 
-    codeGenDir.listFiles()
-      ?.forEach {
-        check(it.deleteRecursively()) {
-          "Could not clean file: $it"
+    if (compilationRound == 1) {
```

Comment:
If I'm reading correctly, wouldn't this still behave the same as the existing behavior? Because we already have the `didRecompile` flag that should prevent us getting to this point after the first invocation of `analysisCompleted`